### ISSUE TITLE
Import passman JSON error "Cannot read property 'toLowerCase' of undefined"

### ISSUE THREAD

**Issue by wouterVE** [marius-wieschollek/passwords#326]
### System Information
**Server:**
```
 Put the output of "./occ passwords:system:report" here or fill out the fields below

 Nextcloud Version: 19.0.4
 PHP Version: 7.3
 Database and Version: MariaDB 10.3.24
 Server OS and Version: Debian 10
 Passwords Version: 2020.12.1
```

**Client:**
```
 Browser and Version: Chrome & Firefox & Opera
 Client OS and Version: Ubuntu & Windows
```

### Steps to reproduce
1. Import .json file of passman database backup
2. Got the following error:

```
TypeError: Cannot read property 'toLowerCase' of undefined
    at Function._processTags (ImportManager.38d84420e0d791c8ffcd.js:1)
    at async Function.processJson (ImportManager.38d84420e0d791c8ffcd.js:1)
    at async Function._convertInputData (ImportManager.38d84420e0d791c8ffcd.js:1)
    at async c.importDatabase (ImportManager.38d84420e0d791c8ffcd.js:1)
(anonymous) @ BackupSection.38d84420e0d791c8ffcd.js:1
Promise.catch (async)
importDb @ BackupSection.38d84420e0d791c8ffcd.js:1
async function (async)
importDb @ BackupSection.38d84420e0d791c8ffcd.js:1
qt @ app.js?v=616b0586-51:43
r @ app.js?v=616b0586-51:43
qt @ app.js?v=616b0586-51:43
t.$emit @ app.js?v=616b0586-51:43
fireEvent @ app.js?v=616b0586-51:1
click @ app.js?v=616b0586-51:1
qt @ app.js?v=616b0586-51:43
r @ app.js?v=616b0586-51:43
Zn.a._wrapper @ app.js?v=616b0586-51:43

```

### Expected result
Sucesfull import of my passwords from passman. 

### Actual result
The error mentioned above

### Nextcloud log
No relevant information

### Additional information
Exporting my Passman database as a .csv file did work to import on passwords though.
I've had many difficulties to export my passman database, needing to use the browsers console to add some java script to eventually manage te succesfully export my database. Maybe the exported .json is not 100% correct (though, I could import it in Passman, so I don't know where the fault is Passman or passwords..)

**Comment by marius-wieschollek** [comment]
Thanks for letting me know about this. I wasn't able to reproduce the issue with test data created on the latest version of Passman. From the error message i can see that something must have been wrong with the format of a tag.

Can you take a look at your .json file and search for the tag properties of the passwords? 
The should look like this: `"tags":[]` if the password had no tag or `"tags":[{"text":"tag1"},{"text":"tag2"}]` if the password had tags.
It would be great to know if there are any tags where the format is something other than `{"text":"tag label"}` or if there are maybe tags where the label is missing like `{"text":undefined}`.

**Comment by wouterVE** [comment]
You're welcome. I'll look it up for you. 

One practical question: the JSON file is now formatted in one single line which makes it hard to read. Is there a simple tool or command line (Ubuntu/Windows) to convert this to human-readable format (I know there are sites to re-format this but as this is sensitive data I don't want to upload this :-))

**Comment by marius-wieschollek** [comment]
If you have python installed, you can use `python -m json.tool yourfile.json > formatted.json`.
If you have node installed, you can use `node -e "console.log(JSON.stringify(JSON.parse(require('fs') \
      .readFileSync(process.argv[1])), null, 4));" yourfile.json > formatted.json`.

Otherwise you can use the browser console (press F12) on a blank tab.
```
let data = <here the contents of your json file>;
console.log(JSON.stringify(data, null, 4));
```

**Comment by wouterVE** [comment]
Hello,
Thanks for your suggestions, `python` did the job. 
I've found one tag with a very long random string:
```
 "tags": "eyJpdiI6IkdnWGI1MDNJY0V4c25UYTZYL0ZKdFEiLCJ2IjoxLCJpdGVyIjoxMDAwLCJrcyI6MjU2LCJ0cyI6NjQsIm1vZGUiOiJjY20iLCJhZGF0YSI6IiIsImNpcGhlciI6ImFlcyIsInNhbHQiOiJ2Ny9QVUNid2JPUSIsImN0IjoiZ1BBNnVwRDhzUXlmOXhQdTk4MFFFanltMHNQSHBmN1lDU3cifQ==",
```
Possibly this is the cause for the problem above? 

Now, is there a way I can test this without overwriting my current passwords database?

kr

wouter

**Comment by marius-wieschollek** [comment]
That's interesting. This looks like a base64 encoded JSON object containing encrypted tags.
No idea how this ended up in the export, but the best thing i can do with it is to ignore it.
I will include patches for this in the next update. Lets hope the other properties are decrypted as they should.

If you want to import the passwords without changing the current ones, you can set the conflict mode to "Create new entry" to create new items instead of updating the existing ones. But that leaves you with duplicates for every item.
I would recommend using a test account or making a [database backup](https://git.mdns.eu/nextcloud/passwords/wikis/Users/Import/Import-from-backup) before testing and then resetting everything in the settings and restoring the backup.

**Comment by wouterVE** [comment]
I've checked my passman manager and for the record in question I do see that there are lots of small gray boxes where the tags would normally go:

![image](https://user-images.githubusercontent.com/29882487/104221684-866f9880-5441-11eb-8cbc-d255e5530134.png)

If I edit the record, It doesn't save a tag when I enter it. So guess It's a bug from Passman after all. I'll check tommorow if I can test it without this tag in the json export with a new NC test account.

**Comment by marius-wieschollek** [comment]
Yes, that seems logical due to a quirk in JavaScript. In JS, you can iterate over an array and get each item that way. If you try the same with a string, you will get each letter in that string. If you then try to read the property "text" of that letter, you will get undefined instead of an error.

I will add a check if the tags are a string instead of an array for the next update.

**Comment by wouterVE** [comment]
Hello, 
I can confirm that I am able to import the .json after I deleted the tag with the very long string.
FYI: in the .csv file the tag looked like this:
```
[,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,]
```

Maybe it's a good idea to implement a check as the export function of passman is verry buggy I have noticed.

**Comment by marius-wieschollek** [comment]
This should be fixed with 2021.1.2
